README.md

```diff
@@ -43,9 +43,7 @@ The file `example/interactive_hasher.cpp` has more detailed information.
 ## Generating SHA256 hash from a binary file
 
 ```c++
-std::ifstream ifs("file.txt", std::ios::binary);
-std::vector<unsigned char> hash(picosha2::k_digest_size);
-picosha2::hash256(std::istreambuf_iterator<char>(ifs), std::istreambuf_iterator<char>(), hash.begin(), hash.end());
+std::string hex_str = hash256_from_ifstream(std::ifstream f);
```

Comment:
Please change the function signature as `template<OutIter> hash256(std::ifstream& ifs, OutIter first, OutIter last)`